### PR TITLE
test: Add a simple test for /todo

### PR DESCRIPTION
**Pull request by jelly**
This add two tests for the todo view for incompleted and scheduled
advisories.

The rest should also be tested, but this should be a nice start.

### CODE REVIEW

**Comment by jelly** [comment]
@SantiagoTorres Yes, we should determine below which h2 the text is shown, but that would involve parsing HTML of which I am a bit reluctant. Maybe @anthraxx has a brilliant idea?

**Comment by anthraxx** [comment]
we can split up TODO like the other things to provide json API for logged in users. that way we can parse the json (like the stats page is doing)

**Comment by jelly** [comment]
@anthraxx wouldn't be bad, since then we can also make a irc bot which bugs when a package is bumped ;-)

**Comment by jelly** [comment]
Hmm that's not even required, we can just split up retrieving the todo data and the request. then just test get_todo_data. A feed would be nice to have too.

File: app/view/todo.py [hunk 1]
```diff
@@ -104,6 +103,12 @@ def todo():
         'bumped_groups': bumped_groups,
         'orphan_issues': orphan_issues
     }
+
+
+@app.route('/todo', methods=['GET'])
```
Comment (anthraxx) [inline app/view/todo.py]: can we get a json endpoint like all other endpoints have?

File: test/test_todo.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from flask import url_for
+
+from .conftest import create_package, create_group, create_issue, create_advisory, DEFAULT_GROUP_ID, DEFAULT_ADVISORY_ID, DEFAULT_ISSUE_ID
+from app.model.enum import Publication, Severity, Remote
+from app.model.cve import issue_types
+from app.view.todo import get_todo_data
+
+
+def todo_data(todo_data, expected):
```
Comment (anthraxx) [inline test/test_todo.py]: can we name this something like assert_todo_date so we know what the function actually does?

File: test/test_todo.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    data = get_todo_data()
+    assert data['unknown_issues']
+    assert data['unknown_issues'][0].id == DEFAULT_ISSUE_ID
+
+
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1])
+def test_orphan_issues_(db, client):
+    data = get_todo_data()
+    assert data['orphan_issues']
+    assert data['orphan_issues'][0].id == DEFAULT_ISSUE_ID
```
Comment (anthraxx) [inline test/test_todo.py]: can you add a test calling the HTML endpoint with everything set up that all test cases/categories show some entries? incomplete/scheduled/unhandled advisories unknown and orphan issues? then check if status code is 200. This way we can ensure there are not serious regressions while displaying the page (no need to check content, just status)

Comment (jelly) [reply test/test_todo.py]: That's something I was also planning on doing, but first wanted to handle all the separate cases.

File: app/view/todo.py (hunk 1 above)
Comment (jelly) [reply app/view/todo.py]: Sure, do we want to split them based on the 'todo type', so scheduled, incomplete advisories?

File: test/test_todo.py (hunk 2 above)
Comment (jelly) [reply test/test_todo.py]: That's the plan, this PR is mostly work in progress still.

**Comment by anthraxx** [comment]
I would say let's just have a single endpoint for todo to be consistent with the web ui (just add Jason request part).
Whoever is interested in just one section can easily use that part of the dictionary from the json.
If you meant that, yes go ahead and I would vote for a similar structure just bit more logically grouped.
Like advisory -> [incomplete, pending...], issue -> [ orphan]...
Think that makes a better structure then advisory_pending. Take a look at the stats PR

**Comment by jelly** [comment]
@anthraxx Rebased this all and wondering how we want to show all the data. I've changed it to your suggestion with advisories, groups, issues. But the groups/advisories have different data so that's inconsistent and I think we want it to be the same for advisories and groups right?

I've this is ironed out I'll add tests (I promise)

**Comment by anthraxx** [comment]
applied via ab037849469e29201f3e24ac0713403485a4e422
thx

**Pull request closed without merging**